src/vmspawn/vmspawn-settings.h

```diff
@@ -0,0 +1,22 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include <errno.h>
+#include <stdint.h>
+
+typedef enum ConfigFeature {
+        CONFIG_FEATURE_AUTO,
+        CONFIG_FEATURE_ENABLED,
+        CONFIG_FEATURE_DISABLED,
+        _CONFIG_FEATURE_MAX,
+        _CONFIG_FEATURE_INVALID = -EINVAL,
+} ConfigFeature;
```

Comment:
so across our codebase we have these "tristate" or sometimes called "extended booleans", i.e. booleans which are true/yes/1 or false/no/0, but also know a third state invalid/unset/auto. We encode these as simple `int` values, with a value < 0 as unset/invalid/auto (typically -1 of course), a value == 0 for false and a value > for true (typically 1 of course). We have some infra for that in place already. It appears to me this new enum should not exist but simply be a tristate.

usually, if people want to select the unset/invalid/auto state explicitly they'd specify an empty string rather than a string such as "auto". 

thus for the kvm switch we should allow `--qemu-kvm=yes`, `--qemu-kvm=no`, --qemu-kvm=` (the latter with an empty assignment to request "auto" behaviour.

typically you#d parse that with:

```c
if (isempty(optarg)) 
        arg_qemu_kvm = -1; 
else {
        r = parse_boolean(optarg);
        if (r < 0) 
                return log_error_errno(r, "…
        arg_qemu_kvm = r;
```

Reply:
Moved to an implementation with a bare int instead of an enum in: 5659535f2e
I have kept the vmspawn-settings.c file despite it only containing
```c
#include "vmspawn-settings.h"
```
because it will be used to house a settings object when loading settings from a file is supported.